airflow/providers/google/cloud/utils/credentials_provider.py

```diff
@@ -332,6 +356,16 @@ def get_credentials_and_project_id(*args, **kwargs) -> tuple[google.auth.credent
     return _CredentialProvider(*args, **kwargs).get_credentials_and_project()
 
 
+def get_id_token_credentials(*args, **kwargs) -> tuple[google.auth.credentials.Credentials]:
```

Comment:
I wonder how much this code has to do with the this code:
https://github.com/apache/airflow/blob/0fd129f33792a95b06c4f4ad1f7ee83dc2f5cef5/airflow/providers/google/common/utils/id_token_credentials.py

Reply:
The two seems indeed related. But `id_token_credentials.py` provide the credentials from [Google application default credentials](https://cloud.google.com/docs/authentication/application-default-credentials).
Whereas the `credential_provider.py` already contains all the code to infer credentials from an Airflow Connection